dashboard/src/reducers/repos.ts

```diff
@@ -46,41 +46,40 @@ const reposReducer = (
 ): IAppRepositoryState => {
   switch (action.type) {
     case getType(actions.repos.receiveRepos):
-      const { repos } = action;
-      return { ...state, isFetching: false, repos, errors: {} };
+      return { ...state, isFetching: false, repos: action.payload, errors: {} };
     case getType(actions.repos.receiveRepo):
-      const { repo } = action;
-      return { ...state, isFetching: false, repo, errors: {} };
+      return { ...state, isFetching: false, repo: action.payload, errors: {} };
     case getType(actions.repos.requestRepos):
       return { ...state, isFetching: true };
     case getType(actions.repos.addRepo):
       return { ...state, addingRepo: true };
     case getType(actions.repos.addedRepo):
-      const { added } = action;
-      return { ...state, addingRepo: false, lastAdded: added, repos: [...state.repos, added] };
+      return {
+        ...state,
+        addingRepo: false,
+        lastAdded: action.payload,
+        repos: [...state.repos, action.payload],
+      };
     case getType(actions.repos.resetForm):
       return { ...state, form: { ...state.form, name: "", namespace: "", url: "" } };
-    case getType(actions.repos.updateForm):
-      const { values } = action;
-      return { ...state, form: { ...state.form, ...values } };
     case getType(actions.repos.showForm):
       return { ...state, form: { ...state.form, show: true } };
     case getType(actions.repos.hideForm):
       return { ...state, form: { ...state.form, show: false } };
     case getType(actions.repos.redirect):
-      return { ...state, redirectTo: action.path };
+      return { ...state, redirectTo: action.payload };
     case getType(actions.repos.redirected):
       return { ...state, redirectTo: undefined };
-    case getType(actions.repos.errorChart):
+    case getType(actions.charts.errorChart):
```

Comment:
this seems like a bug (I know it's correct because it's the same action) but is there any reason for not using `actions.repos.errorChart`?

Reply:
I believe that I removed `actions.repos.errorChart` because it was duplicated. Any reason why you would like to have errorChart inside the repoActions?

Reply:
Hmm, shouldn't they be different? Doesn't this now mean that both the charts and repos reducers will pick up this action and set this? Is that what we desire?

Reply:
Maybe, I do not know. The only thing that I've done is to remove duplicated code.

[This one](https://github.com/kubeapps/kubeapps/pull/679/files#diff-f14bc38d0d5e1089c558504b84144608L27), although referenced by a different variable name, it sends the same action.type `ERROR_CHART`, than https://github.com/kubeapps/kubeapps/pull/679/files#diff-32e6760e59b5e0433a9fae60151e0a8dR19

Both reducers do something with this action, inside their own namespace `repos` or `charts`, I do not know if that's what we want, it is what we had.

If you want me to dig deeper into this let me know, but I think that this will be potentially solved by a general purpose error review and handling.

Reply:
I'm not sure if createAction was creating a special type that was unique even if it had the same type name though? It's possible that getType does something special to ensure that even though they are duplicates, they are different -- although I could also be making stuff up :P

Reply:
>  although I could also be making stuff up :P

You are indeed. It does not do that check afaict. Good point though.

Reply:
Thanks for checking!